### PR TITLE
GroupController

### PR DESCRIPTION
**Pull request by DawidNst**

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/domain/Group.java [hunk 1]
```diff
@@ -23,8 +24,10 @@ public class Group {
     private List<Product> productList = new ArrayList<>();
 
     public Group(String name, String desc) {
-        this.name = name;
+        this.name = String.valueOf(name);
```
Comment (viepovsky) [inline src/main/java/com/kodilla/ecommercee/domain/Group.java:27]: Why use of String.valueOf ?

File: src/main/java/com/kodilla/ecommercee/exception/GroupException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.kodilla.ecommercee.exception;
+
+
+public class GroupException extends Exception {
```
Comment (viepovsky) [inline src/main/java/com/kodilla/ecommercee/exception/GroupException.java]: GroupException should be handled in GlobalError class and it's name should be different than just GroupException so anyone can know what is wrong with the application.

**Review by viepovsky — CHANGES_REQUESTED** [review]
There are also no changes in GroupController class - it must be done

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+public class GroupService {
+
+    private final GroupRepository groupRepository;
+
+    public List<Group> getAllGroups() {
+        return (List<Group>) groupRepository.findAll();
+    }
+
+    public Group getGroupById(Long id) throws GroupException {
+        return groupRepository.findById(id).orElseThrow(() -> new GroupException(
```
Comment (gawrysiakg) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java]: throws GroupNotFoundException will be better. Now we don't know what is the problem with Group.

Comment (DawidNst) [reply src/main/java/com/kodilla/ecommercee/service/GroupService.java]: okk, changed

File: src/main/java/com/kodilla/ecommercee/exception/GroupException.java (hunk 2 above)
Comment (DawidNst) [reply src/main/java/com/kodilla/ecommercee/exception/GroupException.java]: okk, changed

**Review by viepovsky — DISMISSED** [review]
Still no changes in GroupController

File: src/main/java/com/kodilla/ecommercee/controller/GroupController.java [hunk 4]
```diff
@@ -1,44 +1,55 @@
...
+        Group updatedGroup = groupService.updateGroup(group, groupId);
+        return ResponseEntity.ok(groupMapper.mapToGroupDto(updatedGroup));
+    }
+
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<GroupDto> addNewGroup(@RequestBody GroupDto groupDto) {
+        Group group = groupMapper.mapToGroup(groupDto);
+        groupService.saveGroup(group);
+        return ResponseEntity.ok().build();
     }
```
Comment (gawrysiakg) [inline src/main/java/com/kodilla/ecommercee/controller/GroupController.java:48]: If You return ok().build()   -  the method should  be: public ResponseEntity <  Void >  addNewGroup

Comment (DawidNst) [reply src/main/java/com/kodilla/ecommercee/controller/GroupController.java:48]: ok im change now

**Review by MMacina — APPROVED** [review]

**Review by gawrysiakg — APPROVED** [review]

**Pull request merged**